### PR TITLE
REF/ENH: subclass TaoCore with interface methods, add special parsers

### LINKED ISSUE #29: Add logger
Use a logger instead of print statements like:
https://github.com/bmad-sim/pytao/blob/770bc6ff6ee214a4cdd7bb9a6ddff62b2ad21d9c/pytao/interface_commands.py#L26

### PR DESCRIPTION
**Pull request by ken-lauer**
# Changes

* This makes the `Tao` class that users see have interface methods
* This should allow for VSCode and other editors with language servers see the interface methods on the `Tao` object
* Removed runtime method patching
* Adds `pre-commit` configuration for checking the codebase; follow-up PR will do some reformatting if the maintainers are amenable 
* Additionally, closes #29 by adding a `logger.exception` in the noted spot

## Context
I was hoping to get method tab completion in my language server-enabled editor (VSCode, vim + lsp, etc. should all be the same here); interactive Jupyter/IPython have no issue with the current implementation because the instance already exists with the patched-in methods.

### CODE REVIEW

**Review by hhslepicka — APPROVED** [review]
I like this change and how it was implemented.
Thank you for working on that!

**Comment by ChristopherMayes** [comment]
The tests run fine for me locally.

**Comment by ChristopherMayes** [comment]
I see this in the basis notebook:

```python
from pytao import Tao

tao=Tao('-init $ACC_ROOT_DIR/bmad-doc/tao_examples/cesr/tao.init -noplot')   

names = tao.lat_list('*', 'ele.name')

Failed to parse string data. Returning raw value.
Traceback (most recent call last):
  File "[...pytao/pytao/interface_commands.py", line 68](http://localhost:8889/pytao/interface_commands.py#line=67), in __execute
    data = parse_tao_python_data(ret)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "[.../pytao/pytao/tao_ctypes/util.py", line 143](http://localhost:8889/pytao/tao_ctypes/util.py#line=142), in parse_tao_python_data
    dat.update(parse_tao_python_data1(l, clean_key))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "[.../pytao/pytao/tao_ctypes/util.py", line 126](http://localhost:8889/pytao/tao_ctypes/util.py#line=125), in parse_tao_python_data1
    name, type, setable = sline[0:3]
    ^^^^^^^^^^^^^^^^^^^
ValueError: not enough values to unpack (expected 3, got 1)
```

**Comment by ChristopherMayes** [comment]
I tried manually adding a test, which gives the same error when calling directly, but oddly `pytest` doesn't raise anything

```python
def test_lat_list_3():
    tao = Tao("-init $ACC_ROOT_DIR/regression_tests/python_test/cesr/tao.init -noplot")
    
    tao.lat_list(
        ix_uni="1", ix_branch="0", elements="Q*", which="design", who="ele.name"
    )    
```

File: pytao/__init__.py [hunk 1]
```diff
@@ -1,16 +1,28 @@
...
+from .interface_commands import Tao
 
 
 from ._version import get_versions
-__version__ = get_versions()['version']
+
+__version__ = get_versions()["version"]
 del get_versions
+
+__all__ = [
```
Comment (hhslepicka) [inline pytao/__init__.py:21]: Should `__all__` include also `__version__`?

File: pytao/util/parsers.py [hunk 2]
```diff
@@ -10,194 +21,213 @@ def _parse_str_bool(s):
     '0', 'f', 'F' -> Flase
```
Comment (hhslepicka) [inline pytao/util/parsers.py:21]: Not on this PR but there is a typo `Flase` -> `False` 
```suggestion
    '0', 'f', 'F' -> False
```

**Review by hhslepicka — APPROVED** [review]
Amazing work @ken-lauer! I just left a small suggestion to fix a typo (not on this PR) and a question about the `__version__`.

**Pull request merged**